### PR TITLE
Add PolynomialHash Algo

### PR DESCRIPTION
**Pull request by prasad-chavan1**
### Describe your change:
Added Polynomial Rolling Hash Algorithm

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by rohan472000** [comment]
Use `ruff . --fix` to rectify ruff issues, also provide a Wikipedia or any other link for this algo.

**Comment by prasad-chavan1** [comment]
Okay surely...

**Comment by prasad-chavan1** [comment]
> ### Describe your change:
> Added Polynomial Rolling Hash Algorithm
> 
> * [x]  Add an algorithm?
> * [ ]  Fix a bug or typo in an existing algorithm?
> * [ ]  Documentation change?
> 
> ### Checklist:
> * [x]  I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
> * [x]  This pull request is all my own work -- I have not plagiarized.
> * [x]  I know that pull requests will not be merged if they fail the automated tests.
> * [x]  This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
> * [x]  All new Python files are placed inside an existing directory.
> * [x]  All filenames are in all lowercase characters with no spaces or dashes.
> * [x]  All functions and variable names follow Python naming conventions.
> * [x]  All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
> * [x]  All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
> * [x]  All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
> * [ ]  If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

Updated

File: hashes/polynomial_rolling_hash.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""
+Polynomial Rolling Hash Algo -->
+Polynomial rolling hash function is a hash function that uses only multiplications and additions.
```
Comment (rohan472000) [inline hashes/polynomial_rolling_hash.py:3]:
```suggestion
Polynomial rolling hash function is a hash function that uses only multiplications
and additions.
```

**Review by rohan472000 — CHANGES_REQUESTED** [review]

File: hashes/polynomial_rolling_hash.py (hunk 1 above)
Comment (rohan472000) [reply hashes/polynomial_rolling_hash.py:3]: It will rectify ruff issue.

Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py:3]: how can i solve that give me guidance

Comment (rohan472000) [reply hashes/polynomial_rolling_hash.py:3]: Just commit my suggestion.

Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py:3]: can you plz explain again

Comment (rohan472000) [reply hashes/polynomial_rolling_hash.py:3]: Screenshot_20230926-222953
This suggestion to break 3rd line into 2 lines as length of 3rd is more than required.

Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py]: okk sir I'll resolve it by tomorrow..
can I request you to be my mentor for this hacktober fest ?!

Comment (rohan472000) [reply hashes/polynomial_rolling_hash.py]: It is already resolved now as no ruff and pre-conmit check is failing.

Sure :)

Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py]: so now my PR will get merged or not ?
& sir i requested a connection on linkedIn... please accept

Comment (rohan472000) [reply hashes/polynomial_rolling_hash.py]: I and community members will review the PR to ensure it meets repo standards. If any errors are found during the review, we'll make the necessary corrections before considering merging it.

Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py]: ohhh okokk actually I'm contributing 1st time :)
i don't have any idea :(

**Comment by prasad-chavan1** [comment]
## Doctest like -->

>>> s1 = "thealgorithms_python"
>>> s2 = "thealgorithms_python"
>>> s3 = "python_algorithms"

>>> h1 = Hash(s1)
>>> h2 = Hash(s2)
>>> h3 = Hash(s3)

>>> h1 == h2
True
>>> h1 == h3
False

File: hashes/polynomial_rolling_hash.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        hash_so_far = 0
+        p_pow = 1
+        for i in range(self.length):
+            hash_so_far = (
+                hash_so_far + (1 + ord(input_str[i]) - ord("a")) * p_pow
+            ) % self.m
+            p_pow = (p_pow * self.p) % self.m
+        self.hash_value = hash_so_far
+
+    def __eq__(self, other):
```
[the remark this answers is not in the document]
Comment (prasad-chavan1) [reply hashes/polynomial_rolling_hash.py:29]: like

>>> s1 = "thealgorithms_python"
>>> s2 = "thealgorithms_python"
>>> s3 = "python_algorithms"

>>> h1 = Hash(s1)
>>> h2 = Hash(s2)
>>> h3 = Hash(s3)

>>> h1 == h2
True
>>> h1 == h3
False

**Comment by tianyizheng02** [comment]
Thank you for your contribution, but unfortunately we're not accepting new PRs at the moment. We're currently trying to clear our backlog of existing PRs in preparation for Hacktoberfest. If you want to contribute, please wait until after October 1 to do so. See [this discussion](https://github.com/TheAlgorithms/Python/discussions/8912#discussion-5458835) for more info.

**Pull request closed without merging**